apps/qubit/modules/user/actions/loginAction.class.php

```diff
@@ -42,9 +42,19 @@ public function execute($request)
     $this->form->setValidator('next', new sfValidatorString);
     $this->form->setWidget('next', new sfWidgetFormInputHidden);
 
-    $this->form->setValidator('email', new sfValidatorEmail(array('required' => true), array(
-      'required' => $this->context->i18n->__('You must enter your email address'),
-      'invalid' => $this->context->i18n->__('This isn\'t a valid email address'))));
+    // Use string validation if LDAP authentication's used as email not used to log in
+    if ($this->context->user instanceof ldapUser)
```

Comment:
You use `instanceof` here but `get_class` in `menuComponent.class.php`, is that on purpose?